src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -1937,89 +1780,39 @@ make_LedgerConsensus (
-    if (retryAssured)
-        flags = flags | tapRETRY;
+    auto j = app.journal ("LedgerConsensus");
 
-    JLOG (j.debug()) << "TXN "
-        << txn->getTransactionID ()
-        //<< (engine.view().open() ? " open" : " closed")
-        // because of the optional in engine
-        << (retryAssured ? "/retry" : "/final");
-    JLOG (j.trace()) << txn->getJson (0);
+    CanonicalTXSet retriableTxs (set.getHash().as_uint256());
 
-    try
+    for (auto const& item : set)
     {
-        auto const result = apply(app,
-            view, *txn, flags, j);
-        if (result.second)
+        if (! txFilter (item.key()))
+            continue;
+
+        // The transaction isn't in the check ledger, try to apply it
+        JLOG (j.debug()) <<
+            "Processing candidate transaction: " << item.key();
+        std::shared_ptr<STTx const> txn;
+        try
         {
-            JLOG (j.debug())
-                << "Transaction applied: " << transHuman (result.first);
-            return LedgerConsensusImp::resultSuccess;
+            txn = std::make_shared<STTx const>(SerialIter{item.slice()});
         }
-
-        if (isTefFailure (result.first) || isTemMalformed (result.first) ||
-            isTelLocal (result.first))
+        catch (std::exception const&)
         {
-            // failure
-            JLOG (j.debug())
-                << "Transaction failure: " << transHuman (result.first);
-            return LedgerConsensusImp::resultFail;
+            JLOG (j.warn()) << "  Throws";
         }
 
-        JLOG (j.debug())
-            << "Transaction retry: " << transHuman (result.first);
-        return LedgerConsensusImp::resultRetry;
-    }
-    catch (std::exception const&)
-    {
-        JLOG (j.warn()) << "Throws";
-        return LedgerConsensusImp::resultFail;
-    }
-}
-
-void applyTransactions (
-    Application& app,
-    SHAMap const* set,
-    OpenView& view,
-    ReadView const& checkLedger,
-    CanonicalTXSet& retriableTxs,
-    ApplyFlags flags)
-{
-
-    auto j = app.journal ("LedgerConsensus");
-    if (set)
-    {
-        for (auto const& item : *set)
+        if (txn)
         {
-            if (checkLedger.txExists (item.key()))
-                continue;
-
-            // The transaction isn't in the check ledger, try to apply it
-            JLOG (j.debug()) <<
-                "Processing candidate transaction: " << item.key();
-            std::shared_ptr<STTx const> txn;
-            try
-            {
-                txn = std::make_shared<STTx const>(SerialIter{item.slice()});
-            }
-            catch (std::exception const&)
-            {
-                JLOG (j.warn()) << "  Throws";
-            }
-
-            if (txn)
-            {
-                // All transactions execute in canonical order
-                retriableTxs.insert (txn);
-            }
+            // All transactions execute in canonical order
+            retriableTxs.insert (txn);
```

Comment:
Why not put this in the `try` block above? It saves checking `if(txn)`. I suppose it could `throw`, too, which would then get caught and logged instead of propagating out to the caller / thread, but maybe that's a good thing...